keyboards/crkbd/keymaps/stecha/keymap.c

```diff
@@ -0,0 +1,182 @@
+#include QMK_KEYBOARD_H
+
+#ifdef RGBLIGHT_ENABLE
+// Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+extern uint8_t is_master;
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+#define _QWERTY 0
+#define _NUM 1
+#define _SYM 2
+
+enum custom_keycodes { QWERTY = SAFE_RANGE, NUM, SYM, ALT_TAB, DSKP_LEFT, DSKP_RIGHT };
+
+enum macro_keycodes {
+    KC_SAMPLEMACRO,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [_QWERTY] = LAYOUT(KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, LCTL_T(KC_TAB), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, LSFT_T(KC_BSLS), KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT),
+
+    [_NUM] = LAYOUT(KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_BSPC, LCTL_T(KC_TAB), KC_LT, KC_HOME, KC_PGUP, KC_PGDN, KC_END, KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, KC_GT, KC_TAB, LSFT_T(KC_GRV), KC_UNDS, KC_LPRN, KC_DEL, KC_MINS, KC_LBRC, KC_RBRC, KC_EQL, KC_INS, KC_RPRN, KC_PLUS, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT),
+
+    [_SYM] = LAYOUT(KC_ESC, KC_EXLM, KC_AT, KC_HASH, KC_DLR, KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, LSFT_T(KC_GRV), RGB_MOD, RESET, RGB_HUD, RGB_SAD, RGB_VAD, RGB_VAI, RGB_SAI, RGB_HUI, KC_NO, RGB_TOG, KC_RSFT, DSKP_LEFT, NUM, KC_SPC, KC_ENT, SYM, DSKP_RIGHT)};
+
+int RGB_current_mode;
+
+void persistent_default_layer_set(uint16_t default_layer) {
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
+
```

Comment:
Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer